docs/states.md

```diff
@@ -0,0 +1,245 @@
+
+We should always use the warp components for form elements if possible, they handle state styling internally by warp automagic.
+
+
+## Pseudo-elements
+### Before and after
+Style the ::before and ::after pseudo-elements using the `before` and `after` modifiers.
+
+```html
+  <span class="after:ml-16 ..."></span>
+```
+
+It’s worth noting that you don’t really need `::before` and `::after` pseudo-elements for most things in Warp projects — it’s usually simpler to just use a real HTML element.
+
+### Placeholder text
+Style the placeholder text of any input or textarea using the `placeholder` modifier.
+
+html```
+  <input class="placeholder:s-text-placeholder ... " placeholder="Search for anything..." type="text" name="dingbat"/>
+```
+
+### File input buttons
+Style the button in file inputs using the file modifier.
+
+html```
+<input type="file" class="file:mr-16 ... "/>
+```
+
+### List markers
+Style the counters or bullets in lists using the marker modifier.
+
+```html
+<ul role="list" class="marker:s-marker">
+  <li>...</li>
+  ...
+</ul>
+```
+
+### Highlighted text
+Style the active text selection using the selection modifier.
+
+<container>
+<div class="selection:s-bg-primary-active">
+  <p>
+    Space and time are two fundamental concepts that have captivated human imagination for centuries. Space is the vast expanse that surrounds us, both on Earth and beyond, including galaxies, stars, and planets. Time, on the other hand, is the dimension that dictates the order and duration of events in the universe. The relationship between space and time is intricately linked through the concept of spacetime, which emerged from Einstein's theory of relativity. Spacetime is a four-dimensional concept that merges the three dimensions of space with the fourth dimension of time to form a space-time continuum. It is a fundamental concept in modern physics and our understanding of the universe.
+  </p>
+</div>
+</container>
+
+```html
+<div class="selection:s-bg-primary-active">
+  <p>
+    Space and time are two fundamental concepts that have captivated human imagination for centuries. Space is the vast expanse that surrounds us, both on Earth and beyond, including galaxies, stars, and planets. Time, on the other hand, is the dimension that dictates the order and duration of events in the universe. The relationship between space and time is intricately linked through the concept of spacetime, which emerged from Einstein's theory of relativity. Spacetime is a four-dimensional concept that merges the three dimensions of space with the fourth dimension of time to form a space-time continuum. It is a fundamental concept in modern physics and our understanding of the universe.
+  </p>
+</div>
+```
+
+### Dialog backdrops
+Style the backdrop of a native `<dialog>` element using the `backdrop` modifier.
+
+```html
+<dialog class="backdrop:s-bg-backdrop ...">
```

Comment:
We don't have `s-bg-backdrop` either. Should we add that as in the case of `s-marker`?